packages/next/server/render.tsx

```diff
@@ -369,8 +425,14 @@ export async function renderToHTML(
 
   const isServerComponent = !!serverComponentManifest
   const OriginalComponent = renderOpts.Component
+  const serverComponentsInlinedTransformStream =
+    process.browser && isServerComponent ? new TransformStream() : null
   const Component = isServerComponent
-    ? createServerComponentRenderer(OriginalComponent, serverComponentManifest)
+    ? createServerComponentRenderer(
```

Comment:
can we align the condition here `process.browser && isServerComponent` then we don't need the `!` for serverComponentsInlinedTransformStream? not sure if there's case when `process.browser` is `false` here